addon/components/frost-detail-subtabs.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Component definition for the frost-detail-subtabs component
+ */
+import Ember from 'ember'
+const {isEmpty, typeOf} = Ember
+import computed, {readOnly} from 'ember-computed-decorators'
+import {Component} from 'ember-frost-core'
+import {PropTypes} from 'ember-prop-types'
+
+import layout from '../templates/components/frost-detail-subtabs'
+
+export default Component.extend({
+  // == Dependencies ==========================================================
+
+  // == Keyword Properties ====================================================
+
+  layout,
+
+  // == PropTypes =============================================================
+
+  /**
+   * Properties for this component. Options are expected to be (potentially)
+   * passed in to the component. State properties are *not* expected to be
+   * passed in/overwritten.
+   */
+  propTypes: {
+    // options
+    subtabs: PropTypes.arrayOf(PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.object,
+      PropTypes.EmberObject
+    ])).isRequired,
+    selectedSubTab: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.object,
+      PropTypes.EmberObject
+    ]),
+    onSelect: PropTypes.func.isRequired
+    // state
+  },
+
+  /** @returns {Object} the default property values when not provided by consumer */
+  getDefaultProps () {
+    return {
+      // options
+
+      // state
+    }
+  },
+
+  // == Computed Properties ===================================================
+
+  // Wrap selectedTab for the same reasons outlined in the comments for _tabs
+  @readOnly
+  @computed('selectedSubtab')
+  _selectedSubtab (selectedSubtab) {
+    if (!selectedSubtab || typeOf(selectedSubtab) === 'object' || typeOf(selectedSubtab) === 'instance') {
+      return {
+        id: null,
+        label: null
+      }
+    }
+
+    const label = selectedSubtab
+    return {
+      id: label,
+      label,
+      icon: 'view-medium',
+      pack: 'frost'
+    }
+  },
+
+  @readOnly
+  @computed('subtabs.[]')
+  _subtabs (subtabs) {
+    if (isEmpty(subtabs) || typeOf(subtabs[0]) === 'object' || typeOf(subtabs[0]) === 'instance') {
+      // const isMissingProperties = subtabs.some(({id, label, icon, pack}) => {
+      //   return !id || !label
```

Comment:
Please remove the commented code if it is not needed.

Reply:
Done